tests/integration/cluster_fixture.py

```diff
@@ -48,28 +49,87 @@ def _topic_conf(cls, conf=None):
 
         return topic_conf
 
-    def producer(self, conf=None):
+    def producer(self, conf=None, key_serializer=None, value_serializer=None):
         """
         Returns a producer bound to this cluster.
 
         :param dict conf: producer configuration overrides
+        :param Serializer value_serializer: serializer to apply to message key
+        :param Serializer key_serializer:  serializer to apply to message value
+
         :returns: a new producer instance.
         :rtype: confluent_kafka.Producer
         """
+
+        if key_serializer or value_serializer:
```

Comment:
is not None

And do this per-serializer so you don't set None keys in the dict.